### DISCUSSION TITLE
what does `step` mean in `max_steps` ?

### DISCUSSION THREAD

**Discussion by yc1999** [Lightning-AI/pytorch-lightning#12220] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I notice that `step` in PyTorch Lighting can mean `batch` or `optimizer.step()`.  So what does `step` mean in `max_steps` ? @rohitgr7

The following code maybe helpful:

https://github.com/PyTorchLightning/pytorch-lightning/blob/5da065e287b44e2c1fe4f7951003813ed45365c9/pytorch_lightning/loops/epoch/training_epoch_loop.py#L258-L260

https://github.com/PyTorchLightning/pytorch-lightning/blob/5da065e287b44e2c1fe4f7951003813ed45365c9/pytorch_lightning/loops/epoch/training_epoch_loop.py#L323-L331

https://github.com/PyTorchLightning/pytorch-lightning/blob/5da065e287b44e2c1fe4f7951003813ed45365c9/pytorch_lightning/loops/epoch/training_epoch_loop.py#L90-L93

**Comment by rohitgr7** [comment, marked as answer]
it means global_step. So if you have accumulation factors = 2 and max_steps = 10, then the total training batches that will be covered here will be 20 instead of 10. `global_step` indicate total optimization steps.

**Reply by yc1999** [reply]
Thank you, I got it :) BTW, I think the following description maybe ambiguous:

https://github.com/PyTorchLightning/pytorch-lightning/blob/5da065e287b44e2c1fe4f7951003813ed45365c9/pytorch_lightning/loops/epoch/training_epoch_loop.py#L43-L46

I think the "(batches)" should be deleted.😂@rohitgr7

**Reply by rohitgr7** [reply]
ahha right... feel free to send a pr with an update. :)
